frontend/integration-tests/tests/kubevirt/clone.vm.scenario.ts

```diff
@@ -92,118 +88,115 @@ describe('Test clone VM.', () => {
       await vm.action('Clone');
       await wizard.next();
 
-      const clonedVm = new VirtualMachine(clonedVmConf);
-      leakedResources.add(JSON.stringify({name: clonedVm.name, namespace: clonedVm.namespace, kind: 'vm'}));
-      expect(searchYAML(`vm.kubevirt.io/name: ${vm.name}`, clonedVm.name, clonedVm.namespace, 'vm'))
-        .toBeTruthy('Cloned VM should have vm.kubevirt.io/name label.');
-
-      await clonedVm.action('Delete');
-      leakedResources.delete(JSON.stringify({name: clonedVm.name, namespace: clonedVm.namespace, kind: 'vm'}));
+      await withResource(leakedResources, clonedVm.asResource(), async() => {
+        expect(searchYAML(`vm.kubevirt.io/name: ${vm.name}`, clonedVm.name, clonedVm.namespace, 'vm'))
+          .toBeTruthy('Cloned VM should have vm.kubevirt.io/name label.');
+        deleteResource(clonedVm.asResource());
+      });
     }, VM_BOOTUP_TIMEOUT);
 
     it('Clone VM with Container source.', async() => {
       await vm.action('Clone');
       await wizard.next();
 
-      const clonedVm = new VirtualMachine(clonedVmConf);
-      leakedResources.add(JSON.stringify({name: clonedVm.name, namespace: clonedVm.namespace, kind: 'vm'}));
-      expect(searchYAML('kubevirt/cirros-registry-disk-demo', clonedVm.name, clonedVm.namespace, 'vm'))
-        .toBeTruthy('Cloned VM should have container image.');
+      await withResource(leakedResources, clonedVm.asResource(), async() => {
+        expect(searchYAML('kubevirt/cirros-registry-disk-demo', clonedVm.name, clonedVm.namespace, 'vm'))
+          .toBeTruthy('Cloned VM should have container image.');
 
-      await clonedVm.action('Start');
-      await clonedVm.action('Delete');
-      leakedResources.delete(JSON.stringify({name: clonedVm.name, namespace: clonedVm.namespace, kind: 'vm'}));
+        await clonedVm.action('Start');
+        deleteResource(clonedVm.asResource());
+      });
     }, CLONE_VM_TIMEOUT);
+  });
 
-    it('Clone VM with URL source.', async() => {
-      await urlVm.action('Start');
-      await urlVm.action('Clone');
-      await wizard.next();
-
-      const clonedVm = new VirtualMachine({name: `${urlVm.name}-clone`, namespace: urlVm.namespace});
-      leakedResources.add(JSON.stringify({name: clonedVm.name, namespace: clonedVm.namespace, kind: 'vm'}));
-      await clonedVm.action('Start');
-
-      // Check cloned PVC exists
-      const clonedVmDiskName = `${clonedVm.name}-${urlVm.name}-rootdisk-clone`;
-      await browser.get(`${appHost}/k8s/ns/${testName}/persistentvolumeclaims`);
-      await isLoaded();
-      await filterForName(clonedVmDiskName);
-      await resourceRowsPresent();
-
-      // Verify cloned disk dataVolumeTemplate is present in cloned VM manifest
-      expect(searchJSON('spec.dataVolumeTemplates[0].metadata.name', clonedVmDiskName, clonedVm.name, clonedVm.namespace, 'vm'))
-        .toBeTruthy('Cloned VM should have container image.');
+  describe('Test cloning with URL provision method', () => {
```

Comment:
Can you please explain meaning of this test a bit? What does it test (i.e. I do not see any expect) ?

Reply:
The diff is confusing here. This is just a beginning of a new describe block.
I have splitted the original describe block with all test cases into two separate blocks.
The reason is to have better managed resources - to have less resources created at a point in time.

Reply:
ok